src/pyhf/parameters/paramsets.py

```diff
@@ -6,10 +9,18 @@ def __init__(self, **kwargs):
 
 
 class unconstrained(paramset):
-    pass
+    def __init__(self, **kwargs):
+        super(unconstrained, self).__init__(**kwargs)
+        self.constrained = False
+
+
+class constrained_paramset(paramset):
+    def __init__(self, **kwargs):
+        super(constrained_paramset, self).__init__(**kwargs)
+        self.constrained = True
 
 
-class constrained_by_normal(paramset):
+class constrained_by_normal(constrained_paramset):
```

Comment:
comparing this to the constrained_by_poisson, we should use this PR to clean up the code/logic.. they're similar, but not quite the same (re: handling of sigmas/factors and None)